### PR TITLE
feat: Remove unused code from transaction and report service

### PR DESCRIPTION
**Pull request by arjunaj5**
### <samp>🤖 Generated by Copilot at 69ccd9e</samp>

This pull request refactors the `TransactionService` and its spec file by removing obsolete and unused code. It also moves some logic related to expense group summaries to the `ExpenseGroupService` class.
https://app.clickup.com/t/86ctx56yz

### <samp>🤖 Generated by Copilot at 69ccd9e</samp>

> _Some functions were obsolete or unused_
> _So they were removed or diffused_
> _To the `ExpenseGroupService`_
> _Where they have more purpose_
> _And the code style was also improved_

### <samp>🤖 Generated by Copilot at 69ccd9e</samp>

*  Remove `getPaymentModeWiseSummary` and `getCurrenyWiseSummary` functions from `TransactionService` class and their corresponding test cases from `transaction.service.spec.ts` file as they are no longer needed ([link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-712edde8a2bf233766bbc21a32a8166358eca7a00851e1a25af976cff8b780b5L533-L590), [link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L469-L498), [link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L1337-L1403))
*  Remove `isEtxnInPaymentMode` function from `TransactionService` class and its corresponding test case from `transaction.service.spec.ts` file as it is no longer needed ([link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-712edde8a2bf233766bbc21a32a8166358eca7a00851e1a25af976cff8b780b5L793-L812), [link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L1154-L1283))
*  Remove `getPaymentModeForEtxn` and `addEtxnToCurrencyMap` functions from `TransactionService` class and their corresponding test cases from `transaction.service.spec.ts` file as they are no longer needed ([link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-712edde8a2bf233766bbc21a32a8166358eca7a00851e1a25af976cff8b780b5L854-L882), [link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L1154-L1283))
*  Remove trailing commas from the last arguments of various function calls in `transaction.service.spec.ts` file to improve code style and readability ([link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L193-R193), [link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L587-R557), [link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L594-R564), [link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L975-R945), [link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L981-R951), [link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L989-R959), [link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L995-R965), [link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L1001-R971), [link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L1012-R982), [link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L1018-R988), [link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L1029-R999), [link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L1039-R1009), [link](https://github.com/fylein/fyle-mobile-app/pull/2565/files?diff=unified&w=0#diff-f9f7a398e2c2113745a0ca63bb6915be444ffcba34db4710d3f4d1db9c4b7bc5L1301-R1141))

## Clickup
Please add link here

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/services/report.service.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ export class ReportService {
     private approverPlatformApiService: ApproverPlatformApiService,
     private datePipe: DatePipe,
     private launchDarklyService: LaunchDarklyService,
-    private permissionsService: PermissionsService,
+    private permissionsService: PermissionsService
```
Comment (arjunaj5) [inline src/app/core/services/report.service.ts:59]: below are lint fixes

File: src/app/core/services/report.service.ts [hunk 2]
```diff
@@ -603,30 +603,14 @@ export class ReportService {
...
-    } = {
-      params: {},
-    };
-
-    if (orgUserId) {
-      data.params.approver_id = orgUserId;
-    }
-
-    return this.apiService.get('/erpts/' + rptId + '/etxns', data);
-  }
```
Comment (arjunaj5) [inline src/app/core/services/report.service.ts:628]: actual change

File: src/app/core/services/transaction.service.spec.ts [hunk 3]
```diff
@@ -972,33 +942,33 @@ describe('TransactionService', () => {
...
+        'You are about to permanently delete 2 selected expenses.'
       );
     });
   });
 
   describe('getCCCExpenseMessage():', () => {
     it('should return ccc expense message for single ccc expense', () => {
       expect(transactionService.getCCCExpenseMessage(apiExpenseRes, 1)).toEqual(
-        "There is 1 corporate card expense from the selection which can't be deleted. However you can delete the other expenses from the selection.",
+        "There is 1 corporate card expense from the selection which can't be deleted. However you can delete the other expenses from the selection."
```
Comment (hlkavya0213) [inline src/app/core/services/transaction.service.spec.ts:959]: use single quotes

File: src/app/core/services/transaction.service.spec.ts [hunk 4]
```diff
@@ -972,33 +942,33 @@ describe('TransactionService', () => {
...
       );
     });
 
     it('should return ccc expense message for multiple ccc expenses', () => {
       expect(transactionService.getCCCExpenseMessage(expenseList2, 2)).toEqual(
-        "There are 2 corporate card expenses from the selection which can't be deleted. However you can delete the other expenses from the selection.",
+        "There are 2 corporate card expenses from the selection which can't be deleted. However you can delete the other expenses from the selection."
```
Comment (hlkavya0213) [inline src/app/core/services/transaction.service.spec.ts:965]: single quotes

File: src/app/core/services/transaction.service.spec.ts [hunk 5]
```diff
@@ -972,33 +942,33 @@ describe('TransactionService', () => {
...
       );
     });
 
     it('should return ccc expense message for with only ccc expenses selected', () => {
       expect(transactionService.getCCCExpenseMessage(null, 3)).toEqual(
-        "There are 3 corporate card expenses from the selection which can't be deleted. ",
+        "There are 3 corporate card expenses from the selection which can't be deleted. "
```
Comment (hlkavya0213) [inline src/app/core/services/transaction.service.spec.ts:971]: fix it everywhere

**Review by hlkavya0213 — APPROVED** [review]
make sure all references are removed & fix the minor comments

**Pull request merged**